src/Functions/FunctionsStringArray.h

```diff
@@ -239,21 +244,37 @@ class SplitByCharImpl
     Pos end;
 
     char sep;
+    Int64 max_split = -1;
```

Comment:
I would prefer to have optional<UInt64> if -1 is used to indicate that split is not allowed.

Reply:
In Python function split,  default value of max_split is also -1, which meansthere is no limit on the number of splits (all possible splits are made).